### ISSUE TITLE
CarRacing environments can't be pickled

### ISSUE THREAD

**Issue by marho13** [openai/gym#3004]
### Question
Hi, I was trying to use Ray, and when it crashed Multi-processing for running multiple AI agents in gym.
But I received an error which was quite weird, namely that it couldn't find the track in the gym environment.
Looking into it more, it seems that the CarRacing environments are not serializable using the pickle library at least.
Is it supposed to be serializable?

To recreate the problem check the pastebin:
https://pastebin.com/1jw6XC0J

For an example, with multi-threaded environments, here is the pastebin for it:
https://pastebin.com/LNAMbrW2
The error I receive is shown in the image
![Multiprocessing_bug_report](https://user-images.githubusercontent.com/19758684/181731068-8d9eda08-79cb-4655-a7f3-213ca16a90dd.PNG)

Hope to hear from you soon

**Comment by pseudo-rnd-thoughts** [comment]
The issue seems to be related to the Friction detector using a reference to the env where the track has been initialised yet
```python
env = gym.make("CarRacing-v2").unwrapped
print(id(env))
print(id(env.contactListener_keepref.env))

env = pickle.loads(pickle.dumps(gym.make("CarRacing-v2").unwrapped))
print(id(env))
print(id(env.contactListener_keepref.env))
```

Interestingly, bipedal walker and lunar lander have similar implementations but no one has reported issues with it before

**Comment by marho13** [comment]
> The issue seems to be related to the Friction detector using a reference to the env where the track has been initialised yet
> 
> ```python
> env = gym.make("CarRacing-v2").unwrapped
> print(id(env))
> print(id(env.contactListener_keepref.env))
> # These are equal 
> 
> env = pickle.loads(pickle.dumps(gym.make("CarRacing-v2").unwrapped))
> print(id(env))
> print(id(env.contactListener_keepref.env))
> # These are different
> ```
> 
> The solution is probably to move the initialisation of the track to `init` and out of `reset` but this won't fix the issue that the detectors reference different environments
> 
> Interestingly, bipedal walker and lunar lander have similar implementations but no one has reported issues with it before

I think someone may have found the bug previously, as in gym 0.24.1 we find this code in Bipedal:
```python
self.world.contactListener_bug_workaround = ContactDetector(self)
self.world.contactListener = self.world.contactListener_bug_workaround
```
Something very similar is found in LunarLander as well, so I guess for now ill just implement it in the CarRacing environment and see if it works :P Thanks a lot for your help :)

**Comment by pseudo-rnd-thoughts** [comment]
Thanks, the fix were done 6 years old :D but car racing is different as the check is in `init` not `reset` https://github.com/openai/gym/pull/187/files#diff-8f9e342a50c65e66923c4b20efde695f0ce02e0a40873e82ce23f5eb7376b754

**Comment by pseudo-rnd-thoughts** [comment]
@marho13 This should now be fixed by #3025 and available in the next release
